src/main/java/seedu/address/logic/commands/DateCommand.java

```diff
@@ -22,44 +25,93 @@ public class DateCommand extends Command {
+    private final Optional<String> phone;
+    private final Optional<String> email;
     private final Date date;
 
     /**
-     * @param index of the person in the filtered person list to edit the date
+     * @param name of the person to identify person to update
+     * @param phone of the person to identify person to update
+     * @param email of the person to identify person to update
      * @param date of the person to be updated to
      */
-    public DateCommand(Index index, Date date) {
-        requireAllNonNull(index, date);
+    public DateCommand(Optional<String> name, Optional<String> phone, Optional<String> email, Date date) {
+        requireAllNonNull(date);
 
-        this.index = index;
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
         this.date = date;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
+        requireAllNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+
+        // Filter persons based on provided criteria
+        List<Person> matchingPersons = lastShownList.stream()
+                .filter(person -> matches(person))
+                .collect(Collectors.toList());
+
+        // Handle different cases of matches
+        if (matchingPersons.isEmpty()) {
+            throw new CommandException(MESSAGE_NO_PERSON_FOUND);
+        }
+
+        if (matchingPersons.size() > 1) {
+            throw new CommandException(MESSAGE_MULTIPLE_PERSONS_FOUND);
         }
-        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        Person personToEdit = matchingPersons.get(0);
         Person editedPerson = new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
                 personToEdit.getAddress(), personToEdit.getTag(), personToEdit.getAllergy(), date);
         model.setPerson(personToEdit, editedPerson);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
         return new CommandResult(generateSuccessMessage(editedPerson));
     }
 
+    /**
+     * Checks if a person matches the criteria provided in the delete command.
+     */
+    private boolean matches(Person person) {
+        boolean matches = true;
+
+        // If name is provided, check if it matches
+        if (name.isPresent()) {
+            matches &= person.getName().fullName.equalsIgnoreCase(name.get());
+        }
+
+        // If phone is provided, check if it matches
+        if (phone.isPresent()) {
+            matches &= person.getPhone().value.equals(phone.get());
+        }
+
+        // If email is provided, check if it matches
+        if (email.isPresent()) {
+            matches &= person.getEmail().value.equalsIgnoreCase(email.get());
+        }
+
+        return matches;
+    }
```

Comment:
Good use of Optional for handling potential null values, but consider using Optional.map() for cleaner matching logic.